src/v/cluster/rm_stm_types.h

```diff
@@ -9,39 +9,107 @@
 
 #pragma once
 
-#include "cluster/rm_stm.h"
-#include "producer_state.h"
+#include "cluster/types.h"
 #include "reflection/async_adl.h"
 
-using tx_range = cluster::rm_stm::tx_range;
-using prepare_marker = cluster::rm_stm::prepare_marker;
-using abort_index = cluster::rm_stm::abort_index;
-using duration_type = cluster::rm_stm::duration_type;
+namespace cluster::tx {
 
-namespace cluster {
+static constexpr const int8_t abort_snapshot_version = 0;
+static constexpr int8_t prepare_control_record_version{0};
+static constexpr int8_t fence_control_record_v0_version{0};
+static constexpr int8_t fence_control_record_v1_version{1};
+static constexpr int8_t fence_control_record_version{2};
 
-// only retained for snapshot backward compatibility purposes
-// during rollback of an upgrade.
-struct deprecated_seq_entry {
-    static const int seq_cache_size = 5;
+using tx_range = model::tx_range;
+using clock_type = ss::lowres_clock;
+using time_point_type = clock_type::time_point;
+using duration_type = clock_type::duration;
 
-    struct deprecated_seq_cache_entry {
-        int32_t seq{-1};
-        kafka::offset offset;
+struct fence_batch_data {
+    model::batch_identity bid;
+    std::optional<model::tx_seq> tx_seq;
+    std::optional<std::chrono::milliseconds> transaction_timeout_ms;
+    model::partition_id tm;
+};
 
-        bool operator==(const deprecated_seq_cache_entry&) const = default;
-    };
+struct expiration_info {
+    duration_type timeout;
+    time_point_type last_update;
+    bool is_expiration_requested;
+
+    time_point_type deadline() const { return last_update + timeout; }
+
+    bool is_expired(time_point_type now) const {
+        return is_expiration_requested || deadline() <= now;
+    }
+};
+
+struct transaction_info {
```

Comment:
nit: now that this is in tx namespace, the name `transaction_info` has additional gravity :) Can we modify the name to hint that this is about a single partition?

Reply:
ya, this is changing in the next PR, trying to cleanup a lot of stuff in this area.